airbyte-integrations/connectors/source-chargify/source_chargify/source.py

```diff
@@ -0,0 +1,133 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+from urllib.parse import parse_qs, urlparse
+
+import requests
+from requests.auth import HTTPBasicAuth
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.models import SyncMode
+
+
+
+
+# Basic full refresh stream
+class ChargifyStream(HttpStream, ABC):
+
+    def __init__(self, subdomain: str, per_page: int, page: str, *args, **kwargs):
+        super().__init__(**kwargs)
+
+        self._subdomain = subdomain
+        self._per_page = per_page
+        self._page = page
+    
+    @property
+    def url_base(self):
+        return f"https://{self._subdomain}.chargify.com"
+
+    @property
+    def is_first_requests(self)-> bool:
+        return True
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        
+        results = response.json()
+        if results:
+            url_query = urlparse(response.url).query
+            query_params = parse_qs(url_query)
+
+            new_params = {}
+            for param in query_params:
+                if param == "page":
+                    new_params[param] = int(query_params[param][0]) + 1
+                else:
+                    new_params[param] = query_params[param][0]
+
+            return new_params
+        return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        
+        if next_page_token is None and self.is_first_requests is True:
+            self.is_first_requests == False
+            return {"page": self._page, "per_page": self._per_page}
```

Comment:
I don't think `self.is_first_requests` is an useful attribute. `next_page_token == None` means you're making the first request. Moreover, changing a property value could lead to unexpected behavior. If you want to keep `is_first_request` please declare this attribute in the constructor, not in a property.

```suggestion
        if next_page_token is None:
            return {"page": self._page, "per_page": self._per_page}
```

Reply:
More clear sure! changes applied!